plutus-ledger/src/Ledger/Constraints/OnChain.hs

```diff
@@ -40,10 +41,12 @@ checkOwnOutputConstraint
 checkOwnOutputConstraint ctx@ScriptContext{scriptContextTxInfo} OutputConstraint{ocDatum, ocValue} =
     let hsh = V.findDatumHash (Datum $ toBuiltinData ocDatum) scriptContextTxInfo
         checkOutput TxOut{txOutValue, txOutDatumHash=Just svh} =
-            txOutValue == ocValue && hsh == Just svh
+               Ada.fromValue txOutValue >= Ada.fromValue ocValue
```

Comment:
Revert to old implementation when adjusting constraints using manually function as said in another comment.

Reply:
Added another condition below to make it more precise. However, I can't revert it to the old implementation. Here's why. If 10 Ada was locked in the script address, and the user creates a tx to redeem 9 Ada, the wallet emulator balancing will adjust the transaction to only redeem 8 Ada (so that we can leave 2 Ada in the script address), and then the tx validation will fail with the old `checkOwnOutputConstraint`, because the constraint has an output of 1 Ada and the actual tx has an output of 2 Ada.

Reply:
What I meant during the review was a function that adjusts the OutputConstraints, instead of adjusting the unbalanced tx. I'm not sure if that's still a useful thing to do.

Reply:
Ahhh ok. Mmm... I'll check that out. Might be better this way than  to adjust unbalanced tx

Reply:
Tried to write a function which adjusts the TxConstraints instead of UnbalancedTx, but it doesn't seem possible. I had the constraint `mustSpendAtLeast <Guess token>` in the `GameStateMachine` contract. In this scenario, there's a tx output which contains the `Guess` token, but I couldn't modify the `TxConstraints` such that a 2Ada would be associated with the token.